modules/helm-initializer/main.tf

```diff
@@ -26,7 +26,18 @@ data "template_file" "tiller_rbac" {
   }
 }
 
+data "external" "tiller_status" {
+  program = [
+    "bash",
+    "-c",
+    "REPLICAS=$$(kubectl get deploy tiller-deploy -n ${var.tiller_namespace} -o jsonpath='{.status.readyReplicas}'); jq -n --arg replicas \"$$REPLICAS\" '{readyReplicas:$$replicas}'"]
+}
+
 resource "null_resource" "install_tiller" {
+  triggers {
+    tiller_ready_replicas = "${data.external.tiller_status.result.readyReplicas}"
```

Comment:
Thanks Sergey, the tiller deploy is defined as follows: 
```
spec:
...
  replicas: 1
...
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
...
```
so it should not happen normally that there would be more than 1 replica. You're right it can happen in case of an update, but I assume that it should be settled back to 1 on the next TF run. 

I preffered simplicity of implementation in this case, as what would happen if the replicase would be `2` for example, the tiller would just be recreated, which is undesirable, but not really a problem (and as mentioned happens anyway once on the initial cluster creation). Otherwise it can get more cmplicated as there are scenarios when kubectl can't connect or tiller-deply does not exist at all, or it might not have the `readyReplicas` field yet - and after thinking about this I leaned towards keeping it simple.

Reply:
Alright, thanks for the explanation!